private/bufpkg/bufcheck/buflint/internal/buflintcheck/util.go

```diff
@@ -246,21 +246,32 @@ func newMessageCheckFunc(
 func newFieldCheckFunc(
 	f func(addFunc, bufprotosource.Field) error,
 ) func(string, internal.IgnoreFunc, []bufprotosource.File) ([]bufanalysis.FileAnnotation, error) {
-	return newMessageCheckFunc(
-		func(add addFunc, message bufprotosource.Message) error {
-			for _, field := range message.Fields() {
-				if err := f(add, field); err != nil {
-					return err
+	return combine(
+		newMessageCheckFunc(
+			func(add addFunc, message bufprotosource.Message) error {
+				for _, field := range message.Fields() {
+					if err := f(add, field); err != nil {
+						return err
+					}
 				}
-			}
-			// TODO: is this right?
-			for _, field := range message.Extensions() {
-				if err := f(add, field); err != nil {
-					return err
+				for _, field := range message.Extensions() {
+					if err := f(add, field); err != nil {
+						return err
+					}
 				}
-			}
-			return nil
-		},
+				return nil
+			},
+		),
+		newFileCheckFunc(
+			func(add addFunc, file bufprotosource.File) error {
+				for _, field := range file.Extensions() {
+					if err := f(add, field); err != nil {
+						return err
+					}
+				}
+				return nil
+			},
+		),
```

Comment:
Doh! We previously weren't applying lint rules to top-level extensions. Fixed in this branch.